### ISSUE TITLE
Support regex for globals in config

### ISSUE THREAD

**Issue by LunarTwilight** [eslint/eslint#12138]
**The version of ESLint you are using.**
6.2.1

**The problem you want to solve.**
Globals in eslint config do not support regex.

**Your take on the correct solution to problem.**
Allowing support for regex.

**Are you willing to submit a pull request to implement this change?**
No, as I don't know the code needed to implement it.

(I asked about #4158 in the glitter chatroom, and was told to create a new issue about it.)

Could the globals in eslint's config please support regex? As mentioned in #4158 its would make it easier to have globals, as you could use regex to easily set more than one global, rather than having to manually define every single global. Its also taking up unneeded space, and makes the config file harder to read since when you have a lot of globals defined with similar names over and over it starts to become harder to notice globals that are different.

**Comment by LunarTwilight** [comment]
or if regex can't be added, then maybe wildcard support, so you could do `Common*`, which would still work well.

**Comment by kaicataldo** [comment]
I'm personally of the mind that globals should be explicitly defined, but I'm curious what the rest of the team has to say.

**Comment by kaicataldo** [comment]
Also please note that, while we're happy to discuss on this issue, we require an [RFC](https://github.com/eslint/rfcs) for changes to core such as this.

**Comment by LunarTwilight** [comment]
Ah, should I do an RFC now, or wait till its discussed here first?

**Comment by ilyavolodin** [comment]
I also agree with @kaicataldo perspective. I think globals should be defined explicitly and purposefully. I think both RegExp and wildcards might lead to unforeseen issues down the line where they are too inclusive and will not flag variables that are undefined.

**Comment by LunarTwilight** [comment]
Yeah, true, it would make it possible to have false positives, tho regex might not depending on how its done, for example, the way I'd use the regex in my personal config wouldn't require wildcards or etc, since I'm just trying to make it easier to read, since it would be simple regex vs several lines of similar prefixed globals.

**Comment by LunarTwilight** [comment]
like
```yml
globals:
  - Global1: readonly
  - Global2: readonly
  - Global3: readonly
  - Global4: readonly
  - Global5: readonly
  - Global 6: readonly
  - Global 7: readonly
  - Global 8: readonly
  - Global 9: readonly
``` could become just
```yml
globals:
  - /Global[1-9]/: readonly
```
(maybe also use `^` and `$`?)
